src/contrib/torch/tvm_module_wrapper/RuntimeModuleWrapperTVM.cc

```diff
@@ -0,0 +1,281 @@
+
+  std::vector<tvm::runtime::NDArray> input_cache(input_size);
+
+  for (int k = 0; k < input_size; ++k) {
+    auto datum = tvm::contrib::ndarray_from_dlpack(&inputs[k]);  // could have one memory copy
+    input_cache[k] = datum;  // we keep the datum in a vector for future use, otherwise the datum
+                             // will be freed after the loop
+    setter(k, datum);
+  }
+
+  run.CallPacked(tvm::runtime::TVMArgs(tvm_values.data(), tvm_type_codes.data(), input_size),
+                 nullptr);
+
+  for (int k = 0; k < input_size; ++k) {
+    if (tvm_contrib_torch_is_be_copied(&inputs[k]))
+      input_cache[k].CopyTo(&inputs[k].dl_managed_tensor->dl_tensor);
+  }
+}
+
+TVMContribTorchRuntimeModule* tvm_contrib_torch_create_graph_runtime_module(
+    TVMContribTorchRuntimeModule* graph_module, DLManagedTensor* input_example) {
+  tvm::runtime::PackedFunc built_module = graph_module->mod.GetFunction("default");
+  tvm::Device device_info = tvm::contrib::getDeviceInfo(input_example);
+  tvm::runtime::Module runtime_module = built_module(device_info);
+  return new TVMContribTorchRuntimeModule(runtime_module);
+}
+
+size_t tvm_contrib_torch_graph_executor_module_forward(TVMContribTorchRuntimeModule* runtime_module,
+                                                       DLPackTensorExt* inputs, size_t input_size,
+                                                       DLPackTensorExt** outputs) {
+  tvm::runtime::PackedFunc run = runtime_module->mod.GetFunction("run");
+  tvm::runtime::PackedFunc set_input = runtime_module->mod.GetFunction("set_input");
+  tvm::runtime::PackedFunc get_output = runtime_module->mod.GetFunction("get_output");
+  tvm::runtime::PackedFunc get_num_outputs = runtime_module->mod.GetFunction("get_num_outputs");
+
+  for (int k = 0; k < input_size; ++k) {
+    set_input(k, &inputs[k].dl_managed_tensor->dl_tensor);
+  }
+
+  run();
+
+  int64_t output_length = get_num_outputs();
+
+  DLPackTensorExt* outputs_ptr = new DLPackTensorExt[output_length];
+  *outputs = outputs_ptr;
+
+  for (int k = 0; k < output_length; ++k) {
+    tvm::runtime::NDArray results = get_output(k);
+    bool is_bool = results.DataType().is_bool();
+    outputs_ptr[k] = tvm::contrib::create_dlpack_tensor_ext(&results, is_bool);
+  }
+
+  return output_length;
+}
+
+char* tvm_contrib_torch_encode(TVMContribTorchRuntimeModule* runtime_module) {
+  std::string std = tvm::contrib::serialize(runtime_module->mod);
+  char* ret = new char[std.length() + 1];
+  snprintf(ret, std.length() + 1, "%s", std.c_str());
+  return ret;
+}
+
+TVMContribTorchRuntimeModule* tvm_contrib_torch_decode(const char* state) {
+  tvm::runtime::Module ret = tvm::contrib::deserialize(state);
+  return new TVMContribTorchRuntimeModule(ret);
+}
+
+void tvm_contrib_torch_free_runtime_module(TVMContribTorchRuntimeModule* module_ptr) {
+  delete module_ptr;
+}
+
+void tvm_contrib_torch_free_dlpack_tensor_ext_array(DLPackTensorExt* dlpack_ptr) {
+  delete dlpack_ptr;
```

Comment:
I think this needs `delete[]` instead of `delete`. 

https://en.cppreference.com/w/cpp/language/delete
> For the first (non-array) form, ... If expression is anything else, including if it is a pointer obtained by the array form of [new-expression](https://en.cppreference.com/w/cpp/language/new), the behavior is [undefined](https://en.cppreference.com/w/cpp/language/ub).